xmtp/src/builder.rs

```diff
@@ -15,29 +16,36 @@ pub enum ClientBuilderError<PE> {
 
     #[error("Failed to read/write state to persistence: {source}")]
     PersistenceError { source: PE },
+
+    #[error("Error Initalizing Store")]
+    StoreInitialization(#[from] StoreError),
 }
 
 #[derive(Default)]
-pub struct ClientBuilder<P>
+pub struct ClientBuilder<P, S>
 where
     P: Persistence,
+    S: Default,
 {
     network: Network,
     persistence: Option<P>,
     wallet_address: Option<String>,
     account: Option<Account>,
+    store: Option<S>,
```

Comment:
The store type S is specifically un/minimally bounded. Type checking occurs where the object is stored not on the initialization of client